src/config/job-config/jobconfig.interface.ts

```diff
@@ -0,0 +1,85 @@
+import { CreateJobDto } from "../../jobs/dto/create-job.dto";
+import { StatusUpdateJobDto } from "../../jobs/dto/status-update-job.dto";
+import { JobsAuth } from "../../jobs/types/jobs-auth.enum";
+import { JobClass } from "../../jobs/schemas/job.schema";
+
+// Nest Token for CREATE job actions
+export const CREATE_JOB_ACTION_FACTORIES = Symbol(
+  "CREATE_JOB_ACTION_FACTORIES",
+);
+// Nest Token for UPDATE job actions
+export const UPDATE_JOB_ACTION_FACTORIES = Symbol(
+  "UPDATE_JOB_ACTION_FACTORIES",
+);
+
+export interface JobConfigListOptions {
+  configVersion?: string;
+  jobs: JobConfigOptions[];
+}
+
+/**
+ * Encapsulates all responses to a particular job type (eg "archive")
+ */
+export interface JobConfig {
+  jobType: string;
+  configVersion: string;
```

Comment:
This is how it was before. We don't store the top-level object (`JobConfigListOptions` here) so the root `configVersion` has to get copied into each JobConfig section. I'm not sure what happened before if you specified it down a level too. Maybe we should remove it from `JobConfigOptions` entirely, rather than using the top-level field as a default that can still be overridden one level down?

There's potentially a use case where you only update one job type and so only bump one version, but that sounds complicated to track.

Reply:
You are right that this is how it was before, I got confused. I think the confusion came from the fact that we also store `configVersion` in `JobConfigListOptions` but never actually access it from there. From what I see, we extract it directly either from `JobConfig` or `JobConfigOptions`, depending on the case.

We have agreed that we will not cover the use case of updating individually the versions, exactly because it becomes too tricky. There is only a check in the controller that will print an error message when a job's version is not the same as the current running config version - I believe this is enough

Reply:
I've made `JobConfigListOptions.configVersion` required and made sure it always gets passed through to the JobConfig level.

The ajv schema enforces that configVersion can only appear as a top-level field.